### PR TITLE
emails with verbose name supported across the board

### PR DESCRIPTION
**Pull request by LeGast00n**
Signed-off-by: LeGast00n

### CODE REVIEW

**Comment by LeGast00n** [comment]
Hi Selwin. Please check out the commit I've just pushed and let me know what you think.
cheers

File: post_office/validators.py [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
 from django.template import Template, TemplateSyntaxError
 from django.utils.encoding import force_text
 
 from .compat import text_type
 
-
-def validate_email_with_name(value):
-    """
-    Validate email address.
+class FullEmailValidator(EmailValidator):
```
Comment (selwin) [inline post_office/validators.py]: Let's just restore the `validate_email_with_name` function as it was previously and get rid of the `FullEmailValidator` class. It's cleaner and simpler that way. Thanks!

**Comment by LeGast00n** [comment]
Selwin,

Just looking into this again. If I do what you write, I suspect that an address such as "John Smith <john.smith@acme.com>>" (with two trailing '>' characters) would qualify whilst it should not.
What do you think ?

cheers

**Comment by selwin** [comment]
Yeah, you're right. Feel free to update the validation mechanism with the regex based one. I just don't see the point of having a new, class based validator. Thanks!

**Comment by LeGast00n** [comment]
Selwin, the point of having a new, class-based validator is about inheriting from the default (class-based) Email Validator.
Anything wrong with this?
thanks

**Comment by selwin** [comment]
Now that I looked at the code again, it's actually using the built in Django email validator (I thought it wasn't used). Sorry about that.

**Comment by selwin** [comment]
Can you update this PR so it merges cleanly?

File: post_office/validators.py [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
-        start = value.find('<') + 1
-        end = value.find('>')
-        if start < end:
-            recipient = value[start:end]
-    else:
-        recipient = value
+    """
+    def __call__(self, value):
+        try:
+            res = super(FullEmailValidator, self).__call__(value)
```
Comment (selwin) [inline post_office/validators.py]: I wonder if it's more efficient to detect whether display name is present before performing the super call. This way, we can avoid calling Django's builtin email validation twice in the event that email with name is passed in as an argument. Django's builtin validator performs a few regex matches which are relatively expensive compared to simple string based checks.

I don't think we can answer this with no real world usage data though :\

**Comment by LeGast00n** [comment]
Selwin, there come my limits :-/
Can you advise on what I need to do with git to resolve the PR conflicts ?
Thanks

**Comment by selwin** [comment]
Have you pulled the current master branch into your own repo? If it merges cleanly but Github still complains about not being able to merge the PR, it maybe a bug. Try closing this PR and open a new one.

**Comment by LeGast00n** [comment]
What I don't get is that it reads "Only those with write access to this repository can merge pull requests.". Am I really entitled to do it ?
Referring to https://help.github.com/articles/merging-a-pull-request/

**Comment by selwin** [comment]
No, you can't merge this PR since you don't have write access to this repository. However, I can't merge this either since Githun thinks this PR doesn't merge cleanly.

selection_132

I'm going to close this PR, please open a new one and we'll see if I can merge it then.

**Comment by LeGast00n** [comment]
Selwin, I’m litterally lost : can’t fork (I forked it already and it takes me to my existing fork), can’t PR (it asks me to compare 2 forks).

I suck at git…
Can’t I send you the few lines I’ve changed in the 2 amended files so you do it directly on your side ?

**Comment by selwin** [comment]
@LeGast00n sorry for the late reply, yeah just send me a patch and I'll apply it from my end. Or alternatively, if you can point me to a commit in your fork, I can cherry-pick that from your branch so you can still get credited for the commit.

**Comment by LeGast00n** [comment]
Selwin,

Here attached the 3 diffs corresponding to my 3 commits on top of v.1.1.1

Hope this helps.

Kind regards
LA


> Le 19 juin 2015 à 16:44, Louis-Arnaud Iscla <la_iscla@me.com> a écrit :
> 
> Selwin, I’m litterally lost : can’t fork (I forked it already and it takes me to my existing fork), can’t PR (it asks me to compare 2 forks).
> 
> I suck at git…
> Can’t I send you the few lines I’ve changed in the 2 amended files so you do it directly on your side ?
> 
>> Le 19 juin 2015 à 11:48, Selwin Ong <notifications@github.com <mailto:notifications@github.com>> a écrit :
>> 
>> No, you can't merge this PR since you don't have write access to this repository. However, I can't merge this either since Githun thinks this PR doesn't merge cleanly.
>> 
>>  <https://cloud.githubusercontent.com/assets/188658/8250997/e3ad08c6-16a2-11e5-89c1-c08ebc15be12.png>
>> I'm going to close this PR, please open a new one and we'll see if I can merge it then.
>> 
>> —
>> Reply to this email directly or view it on GitHub <https://github.com/ui/django-post_office/pull/88#issuecomment-113449478>.
>> 
> 


diff --git a/post_office/utils.py b/post_office/utils.py
index 2a8e6c8..a681252 100644
--- a/post_office/utils.py
+++ b/post_office/utils.py
@@ -1,8 +1,10 @@
+import re
+
 from django.conf import settings
 from django.core.exceptions import ValidationError
 from django.core.files import File
 from django.core.mail import get_connection
-from django.core.validators import validate_email
+from django.core.validators import EmailValidator
 from django.db.models import Q
 
 try:
@@ -23,6 +25,24 @@
     now = datetime.datetime.now
 
 
+class FullEmailValidator(EmailValidator):
+    """ Simple validator that passes for email addresses bearing a display name
+    i.e. John Smith <john.smith@acme.gov>
+    """
+    def __call__(self, value):
+        try:
+            res = super(FullEmailValidator, self).__call__(value)
+        except ValidationError:
+            try:
+                split_address = re.match(r'(.+) \<(.+@.+)\>', value)
+                display_name, email = split_address.groups()
+                super(FullEmailValidator, self).__call__(email)
+            except AttributeError:
+                raise ValidationError
+
+validate_email = FullEmailValidator()
+
+
 def send_mail(subject, message, from_email, recipient_list, html_message='',
               scheduled_time=None, headers=None, priority=PRIORITY.medium):
     """
diff --git a/post_office/validators.py b/post_office/validators.py
index 994e9dc..5e65738 100644
--- a/post_office/validators.py
+++ b/post_office/validators.py
@@ -34,7 +34,7 @@
 
     for email in value:
         try:
-            validate_email(email)
+            validate_email_with_name(email)
         except ValidationError:
             raise ValidationError('Invalid email: %s' % email, code='invalid')
 

diff --git a/post_office/tests/utils.py b/post_office/tests/utils.py
index c97c9cf..6e912d9 100644
--- a/post_office/tests/utils.py
+++ b/post_office/tests/utils.py
@@ -46,15 +46,16 @@
         # These should validate
         validate_comma_separated_emails(['email@example.com'])
         validate_comma_separated_emails(
-            ['email@example.com', 'email2@example.com', 'email3@example.com']
+            ['email@example.com', 'Alice Bob <email@example.com>',
+             'email2@example.com', 'email3@example.com']
         )
 
         # Should also support international domains
         validate_comma_separated_emails(['email@example.co.id'])
 
-        # These should raise ValidationError
-        self.assertRaises(ValidationError, validate_comma_separated_emails,
-                          ['Alice Bob <email@example.com>'])
+        validate_comma_separated_emails(['Alice Bob <email@example.com>', ])
+
+        # This should raise ValidationError
         self.assertRaises(ValidationError, validate_comma_separated_emails,
                           ['email@example.com', 'invalid_mail', 'email@example.com'])
 
diff --git a/post_office/utils.py b/post_office/utils.py
index a681252..3a4f2f3 100644
--- a/post_office/utils.py
+++ b/post_office/utils.py
@@ -4,7 +4,7 @@
 from django.core.exceptions import ValidationError
 from django.core.files import File
 from django.core.mail import get_connection
-from django.core.validators import EmailValidator
+from django.core.validators import EmailValidator, validate_email
 from django.db.models import Q
 
 try:
@@ -38,9 +38,9 @@
                 display_name, email = split_address.groups()
                 super(FullEmailValidator, self).__call__(email)
             except AttributeError:
-                raise ValidationError
+                raise ValidationError(self.message, code=self.code)
 
-validate_email = FullEmailValidator()
+validate_email = FullEmailValidator(**dict(validate_email.__dict__))
 
 
 def send_mail(subject, message, from_email, recipient_list, html_message='',

diff --git a/post_office/utils.py b/post_office/utils.py
index 3a4f2f3..5be8a16 100644
--- a/post_office/utils.py
+++ b/post_office/utils.py
@@ -1,11 +1,10 @@
-import re
-
 from django.conf import settings
 from django.core.exceptions import ValidationError
 from django.core.files import File
 from django.core.mail import get_connection
-from django.core.validators import EmailValidator, validate_email
 from django.db.models import Q
+
+from .validators import validate_email_with_name
 
 try:
     from django.utils.encoding import force_text
@@ -23,24 +22,6 @@
 except ImportError:
     import datetime
     now = datetime.datetime.now
-
-
-class FullEmailValidator(EmailValidator):
-    """ Simple validator that passes for email addresses bearing a display name
-    i.e. John Smith <john.smith@acme.gov>
-    """
-    def __call__(self, value):
-        try:
-            res = super(FullEmailValidator, self).__call__(value)
-        except ValidationError:
-            try:
-                split_address = re.match(r'(.+) \<(.+@.+)\>', value)
-                display_name, email = split_address.groups()
-                super(FullEmailValidator, self).__call__(email)
-            except AttributeError:
-                raise ValidationError(self.message, code=self.code)
-
-validate_email = FullEmailValidator(**dict(validate_email.__dict__))
 
 
 def send_mail(subject, message, from_email, recipient_list, html_message='',
@@ -181,7 +162,7 @@
 
     for email in emails:
         try:
-            validate_email(email)
+            validate_email_with_name(email)
         except ValidationError:
             raise ValidationError('%s is not a valid email address' % email)
 
diff --git a/post_office/validators.py b/post_office/validators.py
index 5e65738..c956220 100644
--- a/post_office/validators.py
+++ b/post_office/validators.py
@@ -1,28 +1,31 @@
+import re
+
 from django.core.exceptions import ValidationError
-from django.core.validators import validate_email
+from django.core.validators import validate_email as validate_plain_email, EmailValidator
 from django.template import Template, TemplateSyntaxError
 from django.utils.encoding import force_text
 
 from .compat import text_type
 
-
-def validate_email_with_name(value):
-    """
-    Validate email address.
+class FullEmailValidator(EmailValidator):
+    """ Simple validator that passes for email addresses bearing a display name
+    i.e. John Smith <john.smith@acme.gov>
 
     Both "Recipient Name <email@example.com>" and "email@example.com" are valid.
+
     """
-    value = force_text(value)
+    def __call__(self, value):
+        try:
+            res = super(FullEmailValidator, self).__call__(value)
+        except ValidationError:
+            try:
+                split_address = re.match(r'(.+) \<(.+@.+)\>', value)
+                display_name, email = split_address.groups()
+                super(FullEmailValidator, self).__call__(email)
+            except AttributeError:
+                raise ValidationError(self.message, code=self.code)
 
-    if '<' and '>' in value:
-        start = value.find('<') + 1
-        end = value.find('>')
-        if start < end:
-            recipient = value[start:end]
-    else:
-        recipient = value
-
-    validate_email(recipient)
+validate_email_with_name = FullEmailValidator(**dict(validate_plain_email.__dict__))
 
 
 def validate_comma_separated_emails(value):

**Pull request closed without merging**